src/main/java/com/intuit/graphql/orchestrator/xtext/XtextResourceSetBuilder.java

```diff
@@ -57,20 +64,36 @@ public XtextResourceSetBuilder file(String fileName, String file) {
     return this;
   }
 
+  public XtextResourceSetBuilder isFederatedResourceSet(boolean isFederatedResourceSet) {
+    this.isFederatedResourceSet = isFederatedResourceSet;
+    return this;
+  }
+
   /**
    * Build xtext resource set.
    *
    * @return the xtext resource set
    */
   public XtextResourceSet build() {
     graphqlResourceSet = GRAPHQL_INJECTOR.getInstance(XtextResourceSet.class);
-    files.forEach((fileName, content) -> {
+
+    if(isFederatedResourceSet) {
+      String content = FEDERATION_DIRECTIVES + "\n" + StringUtils.join(files.values(), "\n");
+
       try {
-        createGraphqlResourceFromString(content, fileName);
+        createGraphqlResourceFromString(content, "appended_federation");
       } catch (IOException e) {
-        throw new SchemaParseException("Unable to parse file:" + fileName, e);
+        throw new SchemaParseException("Unable to parse file: appended federation file", e);
       }
-    });
+    } else {
+      files.forEach((fileName, content) -> {
+        try {
+          createGraphqlResourceFromString(content, fileName);
```

Comment:
lets create one string of sdl and call createGraphqlResourceFromString only once